### PR TITLE
Update condition code documentation

### LINKED ISSUE #622: HNC: Inconsistent documentation for Condition.Code
[Condition.Code](https://github.com/kubernetes-sigs/multi-tenancy/blob/9f46e6aac110f62fcba9bc4d8d204e68d1760acd/incubator/hnc/api/v1alpha1/hierarchy_types.go#L130) does not contain documentation for [HNSMissing](https://github.com/kubernetes-sigs/multi-tenancy/blob/9f46e6aac110f62fcba9bc4d8d204e68d1760acd/incubator/hnc/api/v1alpha1/hierarchy_types.go#L44), [CannotUpdate](https://github.com/kubernetes-sigs/multi-tenancy/blob/9f46e6aac110f62fcba9bc4d8d204e68d1760acd/incubator/hnc/api/v1alpha1/hierarchy_types.go#L45), [CannotPropagate](https://github.com/kubernetes-sigs/multi-tenancy/blob/9f46e6aac110f62fcba9bc4d8d204e68d1760acd/incubator/hnc/api/v1alpha1/hierarchy_types.go#L46). It also has an extra documentation for [REQUIRED_CHILD_CONFLICT](https://github.com/kubernetes-sigs/multi-tenancy/blob/9f46e6aac110f62fcba9bc4d8d204e68d1760acd/incubator/hnc/api/v1alpha1/hierarchy_types.go#L126), which is not defined [here](https://github.com/kubernetes-sigs/multi-tenancy/blob/9f46e6aac110f62fcba9bc4d8d204e68d1760acd/incubator/hnc/api/v1alpha1/hierarchy_types.go#L40).

### PR DESCRIPTION
**Pull request by yiqigao217**
Add documentation on new codes and delete the obsolete codes.

Tested with "make test".

Fixes #622

### CODE REVIEW

**Comment by yiqigao217** [comment]
@sophieliu15 @adrianchung please feel free to take a look and lmk if anything's not clear in the condition code documentation. Thank you Sophie to bring this up!

File: incubator/hnc/api/v1alpha1/hierarchy_types.go [hunk 1]
```diff
@@ -116,17 +116,22 @@ type Condition struct {
...
-	// - "CRIT_ANCESTOR": a critical error exists in an ancestor namespace, so this namespace is no
+	// - "CritAncestor": a critical error exists in an ancestor namespace, so this namespace is no
 	// longer being updated either.
 	//
-	// - "REQUIRED_CHILD_CONFLICT": this namespace has a required child, but a namespace of the same
-	// name already exists and is not a child of this namespace. Note that the condition is _not_
-	// annotated onto the other namespace; it is considered an error _only_ for the would-be parent
-	// namespace.
+	// - "HNSMissing": this namespace is an owned namespace (created by reconciling an hns instance),
+	// but the hns instance is missing in its owner (referenced in its owner annotation) or the owner
```
Comment (sophieliu15) [inline incubator/hnc/api/v1alpha1/hierarchy_types.go:127]: Will there be more information about which case (i.e., hns instance missing or owner namespace missing) causes this problem when a namespace has this error condition?

File: incubator/hnc/api/v1alpha1/hierarchy_types.go [hunk 2]
```diff
@@ -116,17 +116,22 @@ type Condition struct {
...
+	// namespace is missing.
+	//
+	// - "CannotUpdateObject": this namespace has an error when updating a propagated object. The
+	// affected object will point to the source object.
+	//
+	// - "CannotPropagateObject": this namespace has an error when propagating a source object. The
+	// affected objects will point to all the propagated objects that have an error to update.
```
Comment (sophieliu15) [inline incubator/hnc/api/v1alpha1/hierarchy_types.go:134]: > all the propagated objects that have an error to update.

I am wondering what's the difference between this code and the `CannotUpdateObject` above? From the documentation, they seem to be both about problems when updating propagated objects?

File: incubator/hnc/api/v1alpha1/hierarchy_types.go (hunk 1 above)
Comment (yiqigao217) [reply incubator/hnc/api/v1alpha1/hierarchy_types.go:127]: No, we only look for the hns instance. If it doesn't exist, this condition will be set, no matter the owner namespace exists or not. The two cases here are actually the two possible cases of a missing hns instance.

File: incubator/hnc/api/v1alpha1/hierarchy_types.go (hunk 2 above)
Comment (yiqigao217) [reply incubator/hnc/api/v1alpha1/hierarchy_types.go:134]: Yes, they are two conditions triggered by the same error. However, `CannotPropagateObject` is set on the namespace with the source object and `CannotUpdateObject` is set on the namespace with the propagated object.

Comment (adrianludwin) [reply incubator/hnc/api/v1alpha1/hierarchy_types.go:134]: Maybe: "This object contains an object that couldn't be propagated to one or more of its descendants. The condition's affect objects will include a list of the copies that couldn't be updated."

Comment (yiqigao217) [reply incubator/hnc/api/v1alpha1/hierarchy_types.go]: Thanks all, updated!

**Pull request merged**